src/llmq/quorums_instantsend.h

```diff
@@ -52,23 +52,100 @@ class CInstantSendDb
+     */
     void RemoveInstantSendLock(CDBBatch& batch, const uint256& hash, CInstantSendLockPtr islock);
 
+    /**
+     * This method updates a DB entry for an InstantSend Lock from being not included in a block to being included in a block
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height that the transaction was included at
+     */
     void WriteInstantSendLockMined(const uint256& hash, int nHeight);
+    /**
+     * This method is called when a block gets orphaned for every tx in that block. It marks the ISLocked tx as not included in a block
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height of the block being orphaned
+     */
     void RemoveInstantSendLockMined(const uint256& hash, int nHeight);
+    /**
+     * Marks an InstantSend Lock as archived.
+     * @param batch Object used to batch many calls together
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height that the transaction was included at
+     */
     void WriteInstantSendLockArchived(CDBBatch& batch, const uint256& hash, int nHeight);
+    /**
+     * Archives and deletes all IS Locks which were mined into a block before nUntilHeight
+     * @param nUntilHeight Removes all IS Locks confirmed up until nUntilHeight
+     * @return returns an unordered_map of the hash of the IS Locks and a pointer object to the IS Locks for all IS Locks which were removed
+     */
     std::unordered_map<uint256, CInstantSendLockPtr> RemoveConfirmedInstantSendLocks(int nUntilHeight);
+    /**
+     * Removes IS Locks from the archive if the tx was confirmed 100 blocks before nUntilHeight
+     * @param nUntilHeight the height from which to base the remove of archive IS Locks
+     */
     void RemoveArchivedInstantSendLocks(int nUntilHeight);
+    /**
+     * Checks if the archive contains a certain IS Lock Hash
+     * @param islockHash The hash to check for
+     * @return Returns true if that IS Lock hash is present in the archive, otherwise returns false
+     */
     bool HasArchivedInstantSendLock(const uint256& islockHash);
+    /**
+     * Gets the number of IS Locks which have not been confirmed by a block
+     * @return size_t value of the number of IS Locks not confirmed by a block
+     */
     size_t GetInstantSendLockCount();
 
+    /**
+     * Gets a pointer to the IS Lock based on the hash
+     * @param hash The hash of the IS Lock
+     * @return A Pointer object to the IS Lock, returns nullptr if it doesn't exsist
+     */
     CInstantSendLockPtr GetInstantSendLockByHash(const uint256& hash);
+    /**
+     * Gets an IS Lock hash based on the txid the IS Lock is for
+     * @param txid The txid which is being searched for
+     * @return Returns the hash the IS Lock of the specified txid, returns uint256() if it doesn't exsist
+     */
     uint256 GetInstantSendLockHashByTxid(const uint256& txid);
+    /**
+     * Gets an IS Lock pointer from the txid given
```

Comment:
Should not mention that a pointer is returned. IMHO this is an implementation detail. In the end, it's an object that is returned.

Also...to be honest...I'm really not a big fan of this type of documentation. The meaning of such functions can be completely derived from the method name and the documentation is just rephrasing this. IMHO this is just noise making it harder to read code. This mostly applies to getters/setters, as they really always should do the same (get and set...)...if they ever to do more then this, then documentation is required (or one should reconsider the quality of the code).

Reply:
The LLMQ and PS code in general is very sparsely commented. I agree if you mean just adding Doxygen here w/params and not actually documenting what is being done or how systems work. But I think that short, high-level overviews of things like e.g. in `spork.h` are very helpful to understand what code is doing at a high level and why. The low-level isn't as important as this tends to change more and is more of an implementation detail.

Well commented code is IMO an important first step in creating any kind of spec or protocol agreement, e.g. in case anyone wants to make some other implementation. "Code is protocol" is terrible and ensures that only our own code is going to be run. It doesn't encourage innovation or growth, and makes things harder to read and understand. Esp. w/languages like C++.

Reply:
Fully agree :) I'm not against documentation, I'm against adding documentation to something like `GetName()` which says "Returns the name and the return type is a string".